pkg/ca/x509ca/common_test.go

```diff
@@ -36,3 +47,102 @@ func TestGenerateSerialNumber(t *testing.T) {
 		t.Fatalf("serial number is too large: %v", serialNumber)
 	}
 }
+
+func mustNewTestPublicKey() crypto.PublicKey {
+	priv, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
+	if err != nil {
+		panic(err)
+	}
+	return priv.Public()
+}
+
+func TestMakeCert(t *testing.T) {
+	tests := map[string]struct {
+		Challenge challenges.ChallengeResult
+		WantErr   bool
+		WantFacts map[string]func(x509.Certificate) error
+	}{
+		`Github workflow challenge should have all Github workflow extensions set`: {
```

Comment:
nit: Worth checking that issuer is always set in the GitHub workflow too?

Reply:
Yup the test is there I just didn't add that to the description.

```
`Certifificate should have correct issuer`:     factIssuerIs(`https://token.actions.githubusercontent.com`),
````